src/lib/snarky_js_bindings/lib/snarky_js_bindings_lib.ml

```diff
@@ -2578,14 +2573,8 @@ module Ledger = struct
       Pickles_types.Vector.iter s ~f:(fun x -> ignore (xs##push (field x))) ;
       xs
 
-    let verification_key
-        (vk : (Mina_base.Side_loaded_verification_key.t, Impl.field) With_hash.t)
-        =
-      object%js
-        val hash = field (With_hash.hash vk)
-
-        val data = With_hash.data vk
-      end
+    let verification_key (vk : Mina_base__Verification_key_wire.Stable.V1.t) =
+      Js.string (Pickles.Side_loaded.Verification_key.to_base58_check vk.data)
```

Comment:
I think it might be useful to return the hash (but return the vk as string). The reason is that hashing the vk isn't exposed to snarkyjs yet, and the hash is what represents the vk in the circuit, so is needed to _set_ the vk in a smart contract